feat/register with backend

zo02 opened the pull request:
### Pull Request: Feat/Register with Backend

## Overview
This pull request implements the registration functionality for the frontend to communicate with the backend API. Users can now register using the designated endpoints, enhancing the overall user experience by integrating backend services.

## Changes Made

**Frontend Integration with Backend:**
- Updated the registration view to utilize backend endpoints for user registration.
- Implemented user input validation in the registration form, ensuring all fields are filled out correctly before submission.
- Included a ChoiceBox for user role selection, allowing users to specify their role during registration.

**Login Controller Enhancements:**
- Modified the login controller to facilitate navigation to the registration form.
- This temporary setup will eventually be replaced to direct users to the appropriate view (manager or employee) based on their role after successful login.

## Additional Notes
- The current implementation of opening the registration form upon login is temporary and intended for testing purposes. Future updates will modify this behavior to route users based on their assigned roles.
- Please review the changes made to the registration process and let me know if there are any suggestions or further modifications required!

zo02 committed:
feat/register with backend
Implements the registration funkctionality

blackspyek commented on gui/Test/src/main/java/sample/test/controllers/LoginViewController.java:
```diff
@@ -111,6 +115,24 @@ private void handleSuccessfulLogin(HttpResponse<String> response) {
         loginMessageLabel.setText("Login successful! JWT token stored.");
+
+        try {
+            FXMLLoader loader = new FXMLLoader(Main.class.getResource("register-view.fxml"));
```
Extract it to a new method because its too long to read'

blackspyek commented on gui/Test/src/main/java/sample/test/controllers/LoginViewController.java:
```diff
@@ -111,6 +115,24 @@ private void handleSuccessfulLogin(HttpResponse<String> response) {
+
+            registerStage.show();
+
+            Stage loginStage = (Stage) loginMessageLabel.getScene().getWindow();
```
I would make here another method to close that stage in one line

blackspyek commented on gui/Test/src/main/java/sample/test/dto/RegisterUserDto.java:
```diff
@@ -0,0 +1,39 @@
+    public RegisterUserDto(String username, String password, String role) {
+        this.username = username;
+        this.password = password;
+        this.role = role;
```
So this isnt matching with backend

blackspyek commented on gui/Test/src/main/java/sample/test/dto/RegisterUserDto.java:
```diff
@@ -0,0 +1,39 @@
+
+    // Getters and setters
+    public String getUsername() {
+        return username;
```
Instead of using manual getters and setters use lombok getters and setters

blackspyek commented on gui/Test/src/main/java/sample/test/RegisterViewController.java:
```diff
@@ -39,17 +38,20 @@ public class RegisterViewController implements Initializable {
+    private ChoiceBox<String> roleChoiceBox;
+
+    private String jwtToken;
+    private String[] roles = {"ROLE_USER", "ROLE_MANAGER", "ROLE_EMPLOYEE"};
```
I would use global enum somewhere instead

blackspyek commented on gui/Test/src/main/java/sample/test/RegisterViewController.java:
```diff
@@ -59,57 +61,74 @@ public void closeButtonOnAction(ActionEvent event) {
-        String insertValues = firstname + "','" + lastname + "','" + username + "','" + password + "')";
-        String insertToRegister = insertFields + insertValues;
-
+        RegisterUserDto registerUserDto = new RegisterUserDto(username, password, role);
```
and that's what i've told you on dc, we doesnt provide role in register

blackspyek commented on gui/Test/src/main/java/sample/test/RegisterViewController.java:
```diff
@@ -59,57 +61,74 @@ public void closeButtonOnAction(ActionEvent event) {
+
+            String jwtToken = JwtTokenService.getInstance().getJwtToken();
+
+            if (jwtToken != null && !jwtToken.isBlank()) {
```
that's okay i guess, maybe a little bit too long to read in one method, you could try extract something
also keep in mind that a user with ROLE_MANAGER will only be able to add new Employees

blackspyek reviewed (changes requested):
Overall, this implementation is solid (in several cases need to be repaired), but I recommend refactoring it to enhance clarity and adherence to clean code principles. This way, a new reader will be able to understand the method's purpose at a glance without needing to delve into the logic. Clearer method names, consistent formatting, and well-structured comments could significantly improve readability (these which i commented at)

zo02 committed:
feat/register with backend
Implements the registration funkctionality

zo02 commented:
I cleaned the code and fixed the RegisterUserDto as requested. For now i deleted the roles from this Controller. They will be added to a different controller in the future.

blackspyek reviewed (approved):
## Don't forget to squash
![image](https://github.com/user-attachments/assets/5166ad4f-e199-487c-87dd-fa70a07ac383)
